pkg/resourcemonitor/noderesourcesaggregator.go

```diff
@@ -0,0 +1,274 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourcemonitor
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"time"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	podresourcesapi "k8s.io/kubelet/pkg/apis/podresources/v1"
+
+	"github.com/jaypipes/ghw"
+
+	topologyv1alpha1 "github.com/k8stopologyawareschedwg/noderesourcetopology-api/pkg/apis/topology/v1alpha1"
+)
+
+const (
+	defaultPodResourcesTimeout = 10 * time.Second
+	// obtained these values from node e2e tests : https://github.com/kubernetes/kubernetes/blob/82baa26905c94398a0d19e1b1ecf54eb8acb6029/test/e2e_node/util.go#L70
+)
+
+type nodeResources struct {
+	perNUMACapacity map[int]map[v1.ResourceName]int64
+	// mapping: resourceName -> resourceID -> nodeID
+	resourceID2NUMAID map[string]map[string]int
+	topo              *ghw.TopologyInfo
+}
+
+type resourceData struct {
+	allocatable int64
+	capacity    int64
+}
+
+func NewResourcesAggregator(sysfsRoot string, podResourceClient podresourcesapi.PodResourcesListerClient) (ResourcesAggregator, error) {
+	var err error
+
+	topo, err := ghw.Topology(ghw.WithChroot(sysfsRoot))
+	if err != nil {
+		return nil, err
+	}
+
+	ctx, cancel := context.WithTimeout(context.Background(), defaultPodResourcesTimeout)
+	defer cancel()
+
+	//Pod Resource API client
```

Comment:
A nit, some comments in this PR start with a space after the `//` and other don't, we should follow a single format

Reply:
Sure, will fix.